mypy/meet.py

```diff
@@ -120,6 +120,17 @@ class C(A, B): ...
     if isinstance(s, UnionType):
         return any(is_overlapping_types(t, item)
                    for item in s.relevant_items())
+    if isinstance(t, TupleType) and isinstance(s, TupleType):
+        if t.length() == s.length():
+            if all(is_overlapping_types(ti, si, use_promotions)
+                   for ti, si in zip(t.items, s.items)):
+                return True
+        # TODO: Tuple[A, ...]
```

Comment:
I think that this may work already -- `Tuple[A, ...]` is represented as an `Instance`, and the new check below should handle this (`TupleType` vs. `Instance`). Can you add a test case and remove this comment if I was correct?

Reply:
The case was not handled well, so I added support for it